### DISCUSSION TITLE
How to add some kind of "metadata" during Docker image building to show in source instruction

### DISCUSSION THREAD

**Discussion by peat-psuwit** [pretix/pretix#2028] [Q&A]
As Pretix changes its license to AGPLv3, anyone who modifies Pretix would have to provide the corresponding source. I build a custom Docker image to add a custom font plugin, so I would like to comply.

It's probably enough to just link to my repository in the source instruction (especially since I would deploy the latest version pretty much immediately), but I would love to embed the commit id that a docker image is built from. This would further ensure that the code I linked really corresponds to what is running.

Is it possible to do such a thing in Pretix?

**Comment by raphaelm** [comment]
This is an interesting question and I don't yet have a full answer, but some thoughts.

1. Generally, including versions in a docker image is possible. As a simple approach, you could do a ``git rev-parse --short HEAD > commit.txt`` before your ``docker build`` and then have something like ``COPY commit.txt /commit.txt`` in your docker file, or use docker build args and environment variables.

2. While a very strict interpretation of the AGPL probably would require you to share the exact version and it would be a clear violation to e.g. share an outdated tarball of the source, I think just linking to a git repository is absolutely fine, especially if the git repository is *at least as recent* as the version running and includes all the history. The next pretix version will also include a way to host a list of links to source code at the ``/agpl_source`` location and will recommend to link to all used repositories. So from my standpoint as a copyright holder: Linking to the repo is absolutely fine.

3. Given that we consider linking to the repo to be fine, I don't think linking to or showing the exact version it is something we should offer by default. If there is a critical security vulnerability in pretix or a plugin I'd very much prefer if it wasn't easy to just google installations that did not yet apply the patch.

(Side note, you probably know this, but for other's who are reading here: As long as you do not offer pretix as a service for others, you do not necessarily **have to** share the source of your plugin due to the additional permission we grant.)
